meetings/2022-08-18.md

```diff
@@ -0,0 +1,125 @@
+URL: https://github.com/solid/notifications/issues/97
+
+* RG: RFC. How are we dealing with multiple-resource connections?
+* AC: Lots of ways to address that. Part of the subscription negotiation you can assert a request to listen to arbitrary number of resources. If you then as part of the subscription negotiation, say I want to subscribe to 500 or whatever resources, you can negotiate an access token, to have at least read access to those.. there are different ways to check for that too - different protocols in place. You can even write an implementation that uses id tokens from Solid OIDC. Couple your authz system and websocket system. You can absolutely have 1 connection and listen to multiple resources at the same time.
+* SC: Remind me.. Is that by way of having an array for `topic`?
+* AC: Indeed. Starting point. And then. Just need a way of negotiation is supported by the authorization server or the websocket server itself.
+* SC: ...
+* RG: One or two alternatives are possible. ??? or doing it in band in the connection itself. suppose a resource gets added, do I then have to create a separate connection or is it in band? How will it be managed for addition and removal of resources. At least we have one method. That is clearly stated.
+* AC: The way I'd do this is: first, if websockets makes use of bidirectional flow, or redo this negotiation inside the connection, then websockets is unlike any subscription mechanism. that's what's problematic about this. if however you do this on the subscription level and you want to upgrade, then you can reconnect. then whatever changed between disconnecting and connecting comes through. but for in-band it describes a separate protocol. so we should avoid that.
+* RG: Only websockets is bidirectional.
+* AC: Just because websockets has a mechanism it doesn't mean it is supported in ??? the notifications protocol is trying to be generic. in order to be generic we can't specify / customise for each.
+* RG: This point on generality we may be giving up on efficiency. If for 2.0 protocol, people can look at this. this remains in our memory.
+
+ACTION: RG to create issue to note the cases in which client may want to update its subscriptions or connections.
+
+* AC: For RG to consider, one thing that's important for protocol design is that you can layer. For the kind of thing you are describing re inside the websocket connection, but you could create a service that interacts and calls out a service... inside the service, the client can add/remove/manage whatever mechanism it wants. That service talks to Solid Notifications Protocol service.. and so from a client's perspective, they are doing everything inside the websocket. then you are free at that point because of working on the higher layer of the spec. you can get what you want without having the notifications protocol become more complicated. that's what happens with protocol.. and someone else comes with something that builds on that.
```

Comment:
```suggestion
* AC: For RG to consider, one thing that's important for protocol design is that you can layer. For the kind of thing you are describing re inside the websocket connection, but you could create a service that interacts and calls out a service... Inside the service, the client can add/remove/manage whatever mechanism it wants. That service talks to Solid Notifications Protocol service. From a client's perspective, they are doing everything inside the websocket. You are free at that point because of working on the higher layer of the spec. You can get what you want without having the notifications protocol become more complicated. That's what happens with protocol... and someone else comes with something that builds on that.
```